### PR TITLE
Add weapon categories to suitable tools

### PR DESCRIPTION
**Pull request by RedMisao**
#### Summary
Balance "Add weapon categories to suitable tools"

#### Purpose of change
I noticed some inconsistencies between tool items with enough melee damage to be used as weapons, specifically damage values and the weapon categories they belong to. I asked about both in the discord, and decided to address the latter.

#### Describe the solution
Adding `BATONS`, `HAND_AXES` and `HOOKING_WEAPONRY` to tool items with enough melee damage, according to their closest related point of reference. For example, the heavy_crowbar, which is longer than the crowbar and has a flatter tip lacks `BATONS`, the ny_hook which **is** a hook cannot be used as `HOOKING_WEAPONRY`, the makeshift and primitive hammers are longer than the hammer but are not `BATONS`, the hand_axe lacks `HAND_AXES`, and so on.

#### Describe alternatives you've considered
Not doing this, as this may affect martial arts that use those weapon categories. Then again, I don't see why a halligan, a heavy_crowbar and a heavy_wrench (this one with the same-ish physical characteristics as the normal crowbar) cannot be used as batons, while a crowbar, a pipe and a stick can.
Considered more tools, like the claw_bar, e_tool for `BATONS`, but they're too short (<25 cm).

#### Additional context
Really wanted to increasing the damage of the heavy_wrench using the crowbar as reference but didn't, as there seems to be no point of reference for melee damage.

### CODE REVIEW

**Comment by andrewhr** [comment]
I like this change overall, but question myself if BATONS is the best category. BATONS make me think more about "uniform weight distribution", like the expandable baton, a pipe or even a stick.  OTOH, heavy_wrench I assume will "feel" more like improvised MACES.

I know the category already includes things like hammer, but assuming my point is valid then hammer has a bug. What do you think?

File: data/json/items/tool/entry_tools.json [hunk 1]
```diff
@@ -142,6 +142,7 @@
     "qualities": [ [ "HAMMER", 1 ], [ "PRY", 3 ], [ "COOK", 1 ] ],
     "use_action": [ "CROWBAR" ],
     "melee_damage": { "bash": 45 },
+    "weapon_category": [ "BATONS" ],
```
Comment (Maleclypse) [inline data/json/items/tool/entry_tools.json:145]: Nothing sixty inches long qualifies as a baton.  Specifically anything that sheath's as a spear or requires always_twohand should be in the baton category.
```suggestion
```

**Comment by RedMisao** [comment]
Initially, I chose BATONS instead of MACES, as weapons with the second category are all +60 cm in length, notable examples are baseball bats and the medieval mace, which is 80 cm long.
Yet, your point is valid, if BATONS feel stick-like, that makes the heavy_wrench a small mace. That also makes the hammer(s) closer to MACES than BATONS. I imagine hammers went into BATONS because they're more ergonomic than maces proper.

File: data/json/items/tool/firefighting.json [hunk 2]
```diff
@@ -98,7 +98,8 @@
     "qualities": [ [ "PRY", 4 ], [ "HAMMER", 2 ], [ "DIG", 1 ], [ "PRYING_NAIL", 1 ] ],
     "use_action": [ "CROWBAR" ],
     "flags": [ "DURABLE_MELEE", "BELT_CLIP", "SHEATH_AXE" ],
-    "melee_damage": { "bash": 42, "stab": 5 }
+    "melee_damage": { "bash": 42, "stab": 5 },
+    "weapon_category": [ "BATONS" ]
```
Comment (Maleclypse) [inline data/json/items/tool/firefighting.json:102]:
```suggestion
```
Halligan bars weigh 10-12 pounds.   Ten times the weight of a baton.

File: data/json/items/tool/firefighting.json [hunk 3]
```diff
@@ -220,6 +221,7 @@
     "to_hit": -1,
     "qualities": [ [ "PRY", 3 ], [ "HAMMER", 1 ], [ "DIG", 1 ] ],
     "use_action": [ "CROWBAR" ],
-    "melee_damage": { "bash": 15, "stab": 13 }
+    "melee_damage": { "bash": 15, "stab": 13 },
+    "weapon_category": [ "BATONS", "HOOKING_WEAPONRY" ]
```
Comment (Maleclypse) [inline data/json/items/tool/firefighting.json:225]:
```suggestion
    "weapon_category": [ "HOOKING_WEAPONRY" ]
```
Six times the weight of a baton.

File: data/json/items/tool/workshop.json [hunk 4]
```diff
@@ -694,7 +694,8 @@
     "color": "brown",
     "qualities": [ [ "HAMMER", 2 ] ],
     "flags": [ "NONCONDUCTIVE", "BELT_CLIP" ],
-    "melee_damage": { "bash": 9 }
+    "melee_damage": { "bash": 9 },
+    "weapon_category": [ "BATONS" ]
```
Comment (Maleclypse) [inline data/json/items/tool/workshop.json:698]:
```suggestion
```
One third of the length of batons.

File: data/json/items/tool/workshop.json [hunk 5]
```diff
@@ -980,7 +981,8 @@
     "color": "brown",
     "qualities": [ [ "HAMMER", 2 ] ],
     "flags": [ "BELT_CLIP" ],
-    "melee_damage": { "bash": 14 }
+    "melee_damage": { "bash": 14 },
+    "weapon_category": [ "BATONS" ]
```
Comment (Maleclypse) [inline data/json/items/tool/workshop.json:985]:
```suggestion
```
Same as above

File: data/json/items/tool/workshop.json (hunk 4 above)
Comment (RedMisao) [reply data/json/items/tool/workshop.json:698]: I would like to refer to andrewhr's comment. If the makeshift_hammer and the primitive_hammer are too short for BATONS, then hammer requires a revision too.

File: data/json/items/tool/woodworking.json [hunk 6]
```diff
@@ -519,7 +519,8 @@
     "color": "light_gray",
     "qualities": [ [ "CUT", 1 ], [ "BUTCHER", -50 ], [ "AXE", 1 ], [ "HAMMER", 1 ] ],
     "flags": [ "SHEATH_AXE" ],
-    "melee_damage": { "bash": 7, "cut": 11 }
+    "melee_damage": { "bash": 7, "cut": 11 },
+    "weapon_category": [ "HAND_AXES" ]
```
Comment (TheShadowFerret) [inline data/json/items/tool/woodworking.json:523]:
```suggestion
    "melee_damage": { "bash": 7, "cut": 11 }
```

This has no handle, it is not the form factor of a hand axe, despite its ID name.

Comment (RedMisao) [reply data/json/items/tool/woodworking.json:523]: My bad. I completely missed its description.

File: data/json/items/tool/workshop.json (hunk 4 above)
Comment (kevingranade) [reply data/json/items/tool/workshop.json]: Hammer (I'm assuming a typical carpentry or claw hammer) shouldn't really be considered a baton either no.

**Comment by kevingranade** [comment]
There's no need to go looking for weapons to add to categories, they should be reserved for items that very closely match the properties of weapons in the group, and none of the ones in this PR are.

**Comment by Aerin-of-the-Toast** [comment]
So I've been meaning to write in a more concrete definition for the weapon categories, especially now that there's gameplay ramifications to them outside just martial arts.
For BATONS they should only be of a certain length (in the 40-70cm range I'd say), only be used one-handed, and have a uniform weight distribution or balanced closer towards the hilt. Fast, whippy, short-ish one-handed blunt weapons. The hammer being considered a baton is what's wrong, no hammer-shaped anything should be a baton. The stick also shouldn't be a baton; it's too long.
For MACES it'd be any blunt weapon with a heavier head, usable one-handed (if it's only really usable two-handed it'd be in GREATHAMMERS), and is at least 40cm or so in length. Any tool hammer also shouldn't be a mace, since they're too short.
So overall tool hammers and wrenches generally shouldn't be in any weapon category, they're just really short which makes them very awkward weapons to use that aren't really equivalent to any real weapons.

**Pull request closed without merging**